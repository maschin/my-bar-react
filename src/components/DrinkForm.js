import React from 'react';

export class DrinkForm extends React.Component {
    onSubmit = () => {
        this.props.onSubmit();
    };
    render(){
        return (
          <form className="form" onSubmit={this.onSubmit}>
              <div>
                  <button className="button">{this.props.actionTitle}</button>
              </div>
          </form>
        );
    }
}

export default DrinkForm;