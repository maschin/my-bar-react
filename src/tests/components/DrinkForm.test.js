import React from 'react';
import { shallow } from 'enzyme';
import DrinkForm from '../../components/DrinkForm';

test('should render DrinkForm correctly', () => {
    const wrapper = shallow(<DrinkForm actionTitle='Test title' />);
    expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
    const onSubmit = jest.fn();
    const wrapper = shallow(<DrinkForm actionTitle='Test title' onSubmit={onSubmit} />);
    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    });
    expect(onSubmit).toHaveBeenCalledWith();
});