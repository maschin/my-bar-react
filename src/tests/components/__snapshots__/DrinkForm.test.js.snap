// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DrinkForm correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <div>
    <button
      className="button"
    >
      Test title
    </button>
  </div>
</form>
`;
